client/me/security/controller.js

```diff
@@ -47,10 +49,16 @@ export function twoStep( context, next ) {
 }
 
 export function connectedApplications( context, next ) {
-	context.primary = React.createElement( ConnectedAppsComponent, {
-		userSettings: userSettings,
-		path: context.path,
-	} );
+	context.primary = (
+		<>
+			<AsyncLoad
+				require="state/connected-applications/reducer"
+				callback={ reducer => requireReducer( context.store, 'connectedApplications', reducer ) }
+			/>
+			<ConnectedAppsComponent userSettings={ userSettings } path={ context.path } />
```

Comment:
Rendering `<AsyncLoad>` and `<ConnectedAppsComponent>` means that `ConnectedAppsComponent` is rendered and can select and dispatch at a time when the appropriate reducer is not yet loaded.

That's particularly dangerous for action dispatches. These can become very unexpected noops when the reducer and effect handlers that are supposed to react to it are not yet installed.

What if `<AsyncLoad>` accepted children and rendered a placeholder instead of them until an async condition gets fulfilled?
```jsx
<Wait until={ registerReducer( import( 'state/con-apps/reducer' ) ) }>
  <ConnectedAppsComponent />
</Wait>
```
Then the inner component is guaranteed to mount into a fully configured environment.

This places a new burden on the feature developer: to use certain selector or action, they need to know which reducers and effect handlers they depend on and register them accordingly.

It seems that exposing Redux directly as a public API for feature developers reaches some limitations here.